src/Builder/BuilderUtils.php

```diff
@@ -62,10 +62,20 @@ public function addLineBreak(string $breakChar = ' ', int $lines = 1) : self
      * Add a submenu with a name. The name will be displayed as the item text
      * in the parent menu.
      */
-    public function addSubMenu(string $name, CliMenuBuilder $subMenuBuilder = null) : Builder
+    public function addSubMenu(string $id, string $text, CliMenuBuilder $subMenuBuilder = null) : Builder
     {
-        $this->menuItems[]  = $id = 'submenu-placeholder-' . $name;
-
+        if (isset($this->subMenuBuilders[$id])) {
```

Comment:
I guess if you **really** don't want to break BC too much on this, you could set $text to be optionnal (null by default) and then do something like to allow mismatched arguments:
```php
if ($text instanceof CliMenuBuilder) {
    $subMenuBuilder = $text;
    $text = $id;
} elseif (null === $text) {
    $text = $id;
}
```

Not too inclined to do so but that's just a remark.

Reply:
No let's just keep it as it is here.